packages/a11y/package.json

```diff
@@ -26,8 +26,10 @@
 	"types": "build-types",
 	"dependencies": {
 		"@babel/runtime": "^7.11.2",
-		"@wordpress/dom-ready": "file:../dom-ready",
-		"@wordpress/i18n": "file:../i18n"
+		"@wordpress/dom-ready": "file:../dom-ready"
+	},
+	"peerDependencies": {
+		"@wordpress/i18n": "^3.16.0"
```

Comment:
Just to make sure I understand properly, when we work locally on the repo, these will use the local versions right and not install things from npm registry?

Reply:
Because `@wordpress/i18n` is already declared in the root `package.json` as a `file:package/i18n` link, there will be a symlink to `packages/i18n` in `node_modules/@wordpress/i18n`. The package from NPM registry will never be installed there, because that would directly contradict what the root `package.json` says.

NPM 6 (and also Yarn) will never try to install any peer dependency. It only checks if someone else installed it into `node_modules` already, and will check that the semver matches. And because the package is required by the root `package.json`, and the `version` field in the symlinked package's `package.json` matches, it will be happy.

NPM 7 will also find the dependency satisfied by the symlink in root `node_modules`.

Only if we removed the `i18n` dependency from the root `package.json` (or forgot to add a new package there), that's the only case where unwanted things can happen. NPM 6 will issue a warning that peer dependency was not found. And NPM 7 will install it into root `node_modules` from the NPM registry.

Reply:
Is the intention to allow only the same major version of peer dependencies?

Reply:
Yes, in this case, the `a11y` package uses the `3.16` version of the `i18n` API. `3.15` probably lacks some feature that we make use of, so it's not compatible. Version `2` could have completely different API that is not compatible, and `4` might also break everything.